src/extension/tiltify-client.ts

```diff
@@ -0,0 +1,132 @@
+import axios from 'axios';
+import { get } from '../extension/util/nodecg';
+import { tiltifyTokens } from '../extension/util/replicants';
+import { Configschema } from '../types/generated/configschema';
+import {
+  TiltifyTokenResponse,
+  TiltifyCampaignData,
+  TiltifyRawDonation,
+  TiltifyMilestone,
+  TiltifyDonation
+} from '../types/custom/Tiltify-Types';
+
+const nodecg = get();
+const config = (nodecg.bundleConfig as Configschema).tiltify;
+
+export class TiltifyClient {
+  private readonly clientId: string;
+  private readonly clientSecret: string;
+  public readonly campaignId: string;
+  private readonly TOKEN_EXPIRY_BUFFER = 120;
+
+  constructor() {
+    if (!config?.enable) throw new Error('Tiltify integration disabled');
+    if (!config.clientId || !config.clientSecret || !config.campaign) {
+      throw new Error('Missing Tiltify configuration values');
+    }
```

Comment:
```suggestion
    if (config.enable && (!config.clientId || !config.clientSecret || !config.campaign)) {
      throw new Error('Missing Tiltify configuration values');
    }
```

Not sure if throwing an error if that integration is disabled is a good idea. With the checks in `tiltify.ts` I *guess* this wouldn't happen either way, but I still think checking if it's on and if it's missing any required config variables is better.

Reply:
Yeah it really should not happen but I'll remove it